core-to-plc/src/Language/Plutus/CoreToPLC/Compiler/Expr.hs

```diff
@@ -102,26 +102,25 @@ convDataConRef dc =
 
         pure $ constrs !! index
 
-{- Note [Recursive lets]
-We need to define these with a fixpoint. We can derive a fixpoint operator for values
-already.
+isErrorId :: GHC.Id -> Bool
+isErrorId ghcId = ghcId `elem` GHC.errorIds
 
-However, we also need to work out how to encode recursion over multiple values simultaneously.
-The answer is simple - we pass them through as a tuple.
+{- Note [GHC runtime errors]
+GHC has a number of runtime errors for things like pattern matching failures and so on.
 
-Overall, the translation looks like this. We convert this:
+We just translate these directly into calls to error, throwing away any other information.
```

Comment:
Can't get used to it. Throwing away very important information? Weird.

Reply:
Nowhere to put it! I would probably not have designed `error` to carry no data ;)

Reply:
I think it's similar to the printf issue we were talking about yesterday. Throwing away the error makes sense on the actual blockchain, because there is no sensible way of getting it from the node that does the validation back to the user, but it would still be useful for writing the scripts.